File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");
const N = parseInt(input[0]);
const ls = input.slice(1).map(Number);

let stack = [];
let ans = 0;

for (let i of ls) {
  while (stack.length > 0 && stack[stack.length - 1] <= i) {
    stack.pop();
  }
  ans += stack.length;
  stack.push(i);
}

console.log(ans);

File: 백준/Gold/6198. 옥상 정원 꾸미기/옥상 정원 꾸미기.py
import sys
input = sys.stdin.readline
N = int(input())
ls = [int(input()) for _ in range(N)]
stack = []
ans = 0

for i in ls:
    while stack and i >= stack[-1]:
        stack.pop()
    ans += len(stack)
    stack.append(i)

print(ans)
